gcloud/storage/blob.py

```diff
@@ -434,6 +497,9 @@ def upload_from_file(self, file_obj, rewind=False, size=None,
             'User-Agent': connection.USER_AGENT,
         }
 
+        if key:
+            headers.update(self._get_customer_encryption_headers(key))
```

Comment:
You could pass `headers` to `_get_customer_encryption_headers` and then just update it on the fly.